### PR TITLE
Refactor Eventing FI tests for event delivery and tracing check

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Refactor Eventing FI tests for event delivery check

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/16840

### CODE REVIEW

File: tests/fast-integration/eventing-test/assets/eventing-function.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      dependencies: |
+        {
+          "name": "orders",
+          "version": "1.0.0",
+          "dependencies": {
+            "axios": "^0.19.2"
+          }
+        }
+      source: |
+        let receviedEvents = {};
```
Comment (k15r) [inline tests/fast-integration/eventing-test/assets/eventing-function.yaml:42]: can we get some inline documentation on how to use this function? This was one of the main problems with the original version, we should address it here

File: tests/fast-integration/eventing-test/assets/eventing-function.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            } else {
+              console.log("Discarding event with ce-type:", event["ce-type"])
+              console.log("Discarding event with data:", event.data)
+            }
+            return {metadata: metadata};
+          }
+        }
+---
+apiVersion: hydra.ory.sh/v1alpha1
+kind: OAuth2Client
```
Comment (k15r) [inline tests/fast-integration/eventing-test/assets/eventing-function.yaml:102]: why do we need this? we only use in-cluster eventing, so a full blown oauth2 workflow seems unnecessary in a test

File: tests/fast-integration/eventing-test/eventing-test-prep.js [hunk 3]
```diff
@@ -147,6 +155,32 @@ describe('Eventing tests preparation', function() {
     await prepareAssetsWithCompassFlow();
   });
 
+  it('Prepare eventing-sink function', async function() {
+    debug('Preparing EventingSinkFunction');
+    await deployEventingSinkFunction();
+    await waitForEventingSinkFunction();
+  });
+
+  it('Eventing-sink function should be reachable through secured API Rule', async function() {
```
Comment (k15r) [inline tests/fast-integration/eventing-test/eventing-test-prep.js:164]: we actually do not care in the scope of the eventing test
just exposing as some https endpoint without authentication should be enough

**Review by k15r — COMMENTED** [review]
we could remove all dependencies to oauthkeeper and hydra in the eventing version of the fast integration tests

File: tests/fast-integration/eventing-test/assets/eventing-function.yaml (hunk 1 above)
Comment (mfaizanse) [reply tests/fast-integration/eventing-test/assets/eventing-function.yaml:42]: good point, but I will add it in a separate PR.

File: tests/fast-integration/eventing-test/assets/eventing-function.yaml (hunk 2 above)
Comment (mfaizanse) [reply tests/fast-integration/eventing-test/assets/eventing-function.yaml]: I have removed auth for this.

File: tests/fast-integration/eventing-test/eventing-test-prep.js (hunk 3 above)
Comment (mfaizanse) [reply tests/fast-integration/eventing-test/eventing-test-prep.js:164]: I have removed auth for this.

**Review by k15r — APPROVED** [review]

**Review by rakesh-garimella — APPROVED** [review]

**Pull request merged**